Add file transition for /var/lib/arpwatch/.esmtp_queue

zpytela opened the pull request:
Add file transition so that system_mail_t can in the /var/lib/arpwatch directory
with arpwatch_data_t context create .esmtp_queue directory with mail_home_rw_t context.
It is required for the arpwatch service being able to send emails.

Signed-off-by: Zdenek Pytela <zpytela@redhat.com>

bachradsusi commented:
```
policy/modules/contrib/arpwatch.te:94:ERROR 'unknown type mail_home_rw_t used in transition definition' at token ';' on line 2830:
	type_transition system_mail_t arpwatch_data_t:dir mail_home_rw_t ".esmtp_queue";
#line 94
```

zpytela commented:
3 commits merged into one and the commit message updated

wrabcak commented on arpwatch.te:
```diff
@@ -24,6 +24,8 @@ files_pid_file(arpwatch_var_run_t)
 type arpwatch_unit_file_t;
 systemd_unit_file(arpwatch_unit_file_t)
 
+type mail_home_rw_t;
```
SELinux type mail_home_rw_t is declared here: https://github.com/fedora-selinux/selinux-policy-contrib/blob/rawhide/mta.te#L26. This should be removed.

wrabcak commented on arpwatch.te:
```diff
@@ -91,6 +93,8 @@ userdom_dontaudit_use_unpriv_user_fds(arpwatch_t)
 
 mta_send_mail(arpwatch_t)
 
+filetrans_pattern(system_mail_t, arpwatch_data_t, mail_home_rw_t, dir, ".esmtp_queue")
```
For this named_filetrans interface should be used or created if doesn't exists. For more info see:https://github.com/fedora-selinux/selinux-policy-contrib/blob/rawhide/mta.if#L1292

wrabcak reviewed (changes requested):
Added few comments, you're declaring type which already exits and this can be handled by new or updated existing interface.

zpytela committed:
Allow arpwatch send e-mail notifications BZ(1657327)

Label /var/lib/arpwatch/.esmtp_queue as mail_home_rw_t and add file
transition so that system_mail_t can in the /var/lib/arpwatch directory
with arpwatch_data_t context create .esmtp_queue directory with
mail_home_rw_t context.

It is required for the arpwatch service being able to send email notifications
about changes in its database. If esmtp package is installed to provide the sendmail
rpm-capability, the ~arpwatch/.esmtp_queue directory has to have the right label
so that the daemon is able to send e-mails.

Fedoras 29 and 28 are affected.

Signed-off-by: Zdenek Pytela <zpytela@redhat.com>

zpytela commented:
Fixed a problem with unmatching angle brackets.

zpytela commented:
Scratchbuild finished, selinux-policy-3.14.3-17.bz1657327.100.fc30 created.

rpm -qa selinux-policy\*;matchpathcon /var/lib/arpwatch/.esmtp_queue;sesearch -T -s system_mail_t -t arpwatch_data_t -c dir

before:
selinux-policy-devel-3.14.2-44.fc29.noarch
selinux-policy-doc-3.14.2-44.fc29.noarch
selinux-policy-3.14.2-44.fc29.noarch
selinux-policy-targeted-3.14.2-44.fc29.noarch
/var/lib/arpwatch/.esmtp_queue  system_u:object_r:arpwatch_data_t:s0

after:
selinux-policy-targeted-3.14.3-17.bz1657327.100.fc30.noarch
selinux-policy-3.14.3-17.bz1657327.100.fc30.noarch
selinux-policy-doc-3.14.3-17.bz1657327.100.fc30.noarch
selinux-policy-devel-3.14.3-17.bz1657327.100.fc30.noarch
/var/lib/arpwatch/.esmtp_queue  system_u:object_r:mail_home_rw_t:s0
type_transition system_mail_t arpwatch_data_t:dir mail_home_rw_t ".esmtp_queue";

Looks like all changes were applied. Please review again.

wrabcak commented:
LGTM. 

Thanks.
